src/core/gosling-component.tsx

```diff
@@ -54,7 +55,7 @@ export const GoslingComponent = forwardRef<GoslingRef, GoslingCompProps>((props,
     useEffect(() => {
         if (!ref || !hgRef?.current) return;
         const hgApi = hgRef.current;
-        const api = createApi(hgApi, viewConfig, theme);
+        const api = createApi(hgApi, viewConfig, trackInfos, theme);
```

Comment:
Why not:

```suggestion
        const api = createApi(hgApi, viewConfig, trackInfos.current, theme);
```

And then we don't need to "fake" a ref with the api.